service/src/main/java/tds/assessment/services/impl/AssessmentItemBankGenericDataLoaderServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+import tds.assessment.repositories.loader.itembank.TblStimuliRepository;
+import tds.assessment.repositories.loader.itembank.TblStrandRepository;
+import tds.assessment.repositories.loader.itembank.TblSubjectRepository;
+import tds.assessment.services.AssessmentItemBankGenericDataLoaderService;
+import tds.common.Algorithm;
+import tds.testpackage.model.BlueprintElement;
+import tds.testpackage.model.Item;
+import tds.testpackage.model.ItemGroup;
+import tds.testpackage.model.Stimulus;
+import tds.testpackage.model.TestPackage;
+
+import static tds.assessment.model.BlueprintElementTypes.CLAIM_AND_TARGET_TYPES;
+
+@Service
+public class AssessmentItemBankGenericDataLoaderServiceImpl implements AssessmentItemBankGenericDataLoaderService {
+    private final TblSubjectRepository tblSubjectRepository;
+    private final TblItemRepository tblItemRepository;
+    private final TblStimuliRepository tblStimuliRepository;
+    private final TblStrandRepository tblStrandRepository;
+
+    @Autowired
+    public AssessmentItemBankGenericDataLoaderServiceImpl(final TblSubjectRepository tblSubjectRepository,
+                                                          final TblItemRepository tblItemRepository,
+                                                          final TblStimuliRepository tblStimuliRepository,
+                                                          final TblStrandRepository tblStrandRepository) {
+        this.tblSubjectRepository = tblSubjectRepository;
+        this.tblStimuliRepository = tblStimuliRepository;
+        this.tblItemRepository = tblItemRepository;
+        this.tblStrandRepository = tblStrandRepository;
+
+    }
+
+    @Override
+    public void loadSubject(final TestPackage testPackage, final Client client, final String subjectKey) {
+        final TblSubject tblSubject = new TblSubject(testPackage.getSubject(), subjectKey, client.getKey(), Long.parseLong(testPackage.getVersion()));
+        tblSubjectRepository.save(tblSubject);
+    }
+
+    @Override
+    public void loadTblStimuli(final TestPackage testPackage) {
+        final int bankKey = testPackage.getBankKey();
+        final String version = testPackage.getVersion();
+        // Creates a flat list of all stimuli in the test package
+        List<TblStimulus> stimuli = testPackage.getAssessments().stream().flatMap(
+            assessment -> assessment.getSegments().stream().flatMap(segment -> {
+                if (segment.getAlgorithmType().equalsIgnoreCase(Algorithm.FIXED_FORM.getType())) {
+                    return segment.segmentForms().stream()
+                        .flatMap(form -> form.itemGroups().stream()
+                            .filter(itemGroup -> itemGroup.getStimulus().isPresent())
+                            .map(ItemGroup::getStimulus)
+                            .map(stimulus -> mapStimuliToTblStimuli(bankKey, version, stimulus)));
+
+                } else if (segment.getAlgorithmType().contains("adaptive")) {
```

Comment:
Same question here - are there selection algorithms besides "adaptive2"

Reply:
This was basically done for backwards compatibility. I envision "adaptive" and "adaptive2" being synonymous for our purposes. "adaptive" is apparently an older version of the AIR adaptive algorithm that has been deprecated.